### PR TITLE
add citation and zenodo badge

### PR DESCRIPTION
**Pull request by nden**
Adds a Zenodo DOI and a citation file based on the current contributors list 

**Checklist for maintainers**
- [ ] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                  Crawford, Steven and
+                  Law, David and
+                  Gordon, Karl and
+                  Regan, Michael and
+                  Cara, Mihai and
+                  MacDonald, Ken and
+                  Bradley, Larry and
+                  Shanahan, Clare and
+                  Jamieson, William},
+  title        = {JWST Calibration Pipeline1.6.2,
```
Comment (hbushouse) [inline CITATION.cff:28]: should there a whitespace before the version number?

**Review by hbushouse — APPROVED** [review]
Just one minor thing.

**Comment by zacharyburnett** [comment]
Does this warrant a changelog entry so users are aware of the new citation?

File: CITATION.cff (hunk 1 above)
Comment (zacharyburnett) [inline CITATION.cff:28]:
```suggestion
  title        = {JWST Calibration Pipeline 1.6.2,
```

**Review by zacharyburnett — APPROVED** [review]
looks good to me!

File: CITATION.cff (hunk 1 above)
Comment (zacharyburnett) [reply CITATION.cff:28]:
```suggestion
  title        = {JWST Calibration Pipeline 1.6.2,
```

**Comment by jdavies-st** [comment]
FWIW, I updated the release procedures WIKI here so that the `CITATION.cff` file is updated to the current release version and release date. ✅

**Pull request merged**